From patchwork Fri Jan 28 13:10:03 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michel Lespinasse <michel@lespinasse.org>
X-Patchwork-Id: 12728526
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 1C962C433F5
	for <linux-mm@archiver.kernel.org>; Fri, 28 Jan 2022 13:11:00 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 82F306B008A; Fri, 28 Jan 2022 08:10:15 -0500 (EST)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 6E8A36B00A4; Fri, 28 Jan 2022 08:10:15 -0500 (EST)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 99D1A6B008A; Fri, 28 Jan 2022 08:10:14 -0500 (EST)
X-Delivered-To: linux-mm@kvack.org
Received: from forelay.hostedemail.com (smtprelay0199.hostedemail.com
 [216.40.44.199])
	by kanga.kvack.org (Postfix) with ESMTP id 814E06B009F
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 08:10:13 -0500 (EST)
Received: from smtpin06.hostedemail.com (10.5.19.251.rfc1918.com
 [10.5.19.251])
	by forelay02.hostedemail.com (Postfix) with ESMTP id 48C6691E4B
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:13 +0000 (UTC)
X-FDA: 79079729106.06.4EBB877
Received: from server.lespinasse.org (server.lespinasse.org [63.205.204.226])
	by imf30.hostedemail.com (Postfix) with ESMTP id D526E8000C
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:12 +0000 (UTC)
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed;
 d=lespinasse.org; i=@lespinasse.org; q=dns/txt; s=srv-52-ed;
 t=1643375407; h=from : to : cc : subject : date : message-id :
 in-reply-to : references : mime-version : content-transfer-encoding :
 from; bh=hJIKChkB8VHiAv0wjag4cY+zBFGl6h7s8aJLFAJ4t0k=;
 b=Oyga/3lCe/RtyePrIqDmFLWhX0WM4mw25anmtIa6D8dSMSzDkLudB3p1WZ9iuB7b0C2pS
 eHBPvV4iuqF+1dfBQ==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=lespinasse.org;
 i=@lespinasse.org; q=dns/txt; s=srv-52-rsa; t=1643375407; h=from : to
 : cc : subject : date : message-id : in-reply-to : references :
 mime-version : content-transfer-encoding : from;
 bh=hJIKChkB8VHiAv0wjag4cY+zBFGl6h7s8aJLFAJ4t0k=;
 b=UsMUlczIw2jWjOR5SBzHQnYm9vx8Q0Npk09M/ESPvhfFkeTqtcANk3M9DqGLJRlfmqctP
 aXhM1bdRoDv/uV17HzEB/CfGYygE0r5zLawZTdSKm8lOScN6cotkIoMmXwS6NZx8/ndbtDR
 CtLcYcf7Q+yhYzFs766riGUBWIYS8IPqkElBrJV3oOZ2nN9rNQFZGZ4ZnwP9MDSH0X409G6
 UQWDalHyrq+/0D6y2FwuoFRRrSxZyLfr6es3W8xkSvUK0ouTQhRgOm3+PBCokIL5ueLe99n
 eyedRbdNwHfm/c9v4K6IlG4cs1EzJGVjlBsaQy/R+2NNuEoCEKmHM7S4fsJw==
Received: from zeus.lespinasse.org (zeus.lespinasse.org [10.0.0.150])
	by server.lespinasse.org (Postfix) with ESMTPS id 41B2D160ADF;
	Fri, 28 Jan 2022 05:10:07 -0800 (PST)
Received: by zeus.lespinasse.org (Postfix, from userid 1000)
	id 252BB20FB3; Fri, 28 Jan 2022 05:10:07 -0800 (PST)
From: Michel Lespinasse <michel@lespinasse.org>
To: Linux-MM <linux-mm@kvack.org>,
	linux-kernel@vger.kernel.org,
	Andrew Morton <akpm@linux-foundation.org>
Cc: kernel-team@fb.com,
	Laurent Dufour <ldufour@linux.ibm.com>,
	Jerome Glisse <jglisse@google.com>,
	Peter Zijlstra <peterz@infradead.org>,
	Michal Hocko <mhocko@suse.com>,
	Vlastimil Babka <vbabka@suse.cz>,
	Davidlohr Bueso <dave@stgolabs.net>,
	Matthew Wilcox <willy@infradead.org>,
	Liam Howlett <liam.howlett@oracle.com>,
	Rik van Riel <riel@surriel.com>,
	Paul McKenney <paulmck@kernel.org>,
	Song Liu <songliubraving@fb.com>,
	Suren Baghdasaryan <surenb@google.com>,
	Minchan Kim <minchan@google.com>,
	Joel Fernandes <joelaf@google.com>,
	David Rientjes <rientjes@google.com>,
	Axel Rasmussen <axelrasmussen@google.com>,
	Andy Lutomirski <luto@kernel.org>,
	Michel Lespinasse <michel@lespinasse.org>
Subject: [PATCH 32/35] [PATCH v2 32/35] arm64/mm: define
 ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT
Date: Fri, 28 Jan 2022 05:10:03 -0800
Message-Id: <20220128131006.67712-33-michel@lespinasse.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20220128131006.67712-1-michel@lespinasse.org>
References: <20220128131006.67712-1-michel@lespinasse.org>
MIME-Version: 1.0
X-Rspam-User: nil
X-Rspamd-Server: rspam10
X-Rspamd-Queue-Id: D526E8000C
X-Stat-Signature: xku9i8zutkz9mk9qy1djbbcmp11nndg9
Authentication-Results: imf30.hostedemail.com;
	dkim=pass header.d=lespinasse.org header.s=srv-52-ed header.b="Oyga/3lC";
	dkim=pass header.d=lespinasse.org header.s=srv-52-rsa header.b=UsMUlczI;
	spf=pass (imf30.hostedemail.com: domain of walken@lespinasse.org designates
 63.205.204.226 as permitted sender) smtp.mailfrom=walken@lespinasse.org;
	dmarc=pass (policy=none) header.from=lespinasse.org
X-HE-Tag: 1643375412-788441
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

Set ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT so that the speculative fault
handling code can be compiled on this architecture.

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
---
 arch/arm64/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 6978140edfa4..e764329b11a7 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -222,6 +222,7 @@ config ARM64
 	select THREAD_INFO_IN_TASK
 	select HAVE_ARCH_USERFAULTFD_MINOR if USERFAULTFD
 	select TRACE_IRQFLAGS_SUPPORT
+	select ARCH_SUPPORTS_SPECULATIVE_PAGE_FAULT
 	help
 	  ARM 64-bit (AArch64) Linux support.
 

